File: web/src/dtos/Subscription/SubscriptionEditReq.ts
import { Common } from "../../utility";
import { UserRes } from "../User";

export default class SubscriptionEditReq {
  planId?: string = "";
  userId?: string = "";
  activeFrom?: string = Common.dateToString(new Date());
  activeTo?: string = Common.dateToString(new Date());

  user?: UserRes;
}

File: api/Dtos/Subscription/SubscriptionEditReq.cs
using api.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace api.Dtos.Subscription
{
    public class SubscriptionEditReq
    {
        [Required]
        public int? PlanId { get; set; }
        public string? UserId { get; set; }
        public DateTime ActiveFrom { get; set; } = DateTime.MinValue;
        public DateTime ActiveTo { get; set; } = DateTime.MinValue;
    }
}
